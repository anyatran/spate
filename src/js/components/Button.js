import React from 'react'
import PropTypes from 'prop-types'

const Button = ({ text, ...props }) => (
  <button className="button" {...props}>{text}</button>
)

Button.propTypes = {
  text: PropTypes.string
}

export default Button
