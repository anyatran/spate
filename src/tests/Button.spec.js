import React from 'react'
import { mount } from 'enzyme'
import Button from '../js/components/Button'

describe('<Button />', () => {
  it('should render Button', () => {
    const wrapper = mount(<Button text="hello"/>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.props().text).toEqual("hello")
  })
})
