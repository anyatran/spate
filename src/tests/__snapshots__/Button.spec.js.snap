// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button 1`] = `
<Button
  text="hello"
>
  <button
    className="button"
  >
    hello
  </button>
</Button>
`;
